modules/core/src/smithy4s/codecs/FieldSkipCompiler.scala

```diff
@@ -0,0 +1,144 @@
+/*
+ *  Copyright 2021-2025 Disney Streaming
+ *
+ *  Licensed under the Tomorrow Open Source Technology License, Version 1.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *     https://disneystreaming.github.io/TOST-1.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package smithy4s.codecs
+
+import smithy4s.schema.Field
+import smithy4s.schema.Schema
+
+trait FieldSkipCompiler { self =>
+  def compile[A](
+      field: Field[?, A]
+  ): FieldSkipCompiler.ShouldSkip[A]
+
+  def combine(
+      other: FieldSkipCompiler
+  ): FieldSkipCompiler =
+    new FieldSkipCompiler {
+
+      def compile[A](
+          field: Field[?, A]
+      ): FieldSkipCompiler.ShouldSkip[A] = {
```

Comment:
I wonder if encoding this as a `case class FieldSkipCombine(left: ..., right: ...)` wouldn't be a better choice, for debuggability (toString, debugger, profiler)

but maybe it's overkill